torch/ao/quantization/fx/_model_report/model_report_observer.py

```diff
@@ -113,6 +136,75 @@ def _calculate_min_max_stats(self, x_copy):
 
         return x_copy
 
+    def _calculate_percentile_stats(self, x_copy):
+        r"""Calculates and stores the per_channel percentile stats with forward values.
+        Does calculation based on channel axis: self.ch_axis
+
+        Args
+            x_copy: A copy of the forward data
+
+        Returns the passed in x_copy
+        """
+        # get the dimension of the copy
+        x_dim = x_copy.size()
+
+        new_axis_list = [i for i in range(len(x_dim))]  # noqa: C416
+        new_axis_list[self.ch_axis] = 0
+        new_axis_list[0] = self.ch_axis
+        y = x_copy.permute(new_axis_list)
+        # Need to match dtype of min/max because the updates to buffers
+        # are done in place and types need to match for comparisons
+        y = y.to(self.min_val.dtype)
+        y = torch.flatten(y, start_dim=1)
+        y = y.to(self.min_val.dtype)
+
+        # find the percentile values along the axis
+        # we want both 100th percentile and comp_percentile
+        quantiles_list = [self.comp_percentile, 1.00]
+        quantiles_to_find = torch.tensor(quantiles_list, dtype=self.min_val.dtype)
+
+        # find the quantiles
+        comp_quantile = torch.quantile(y, quantiles_to_find[0], dim=self.ch_axis, interpolation="lower")
+        cent_quartile = torch.quantile(y, quantiles_to_find[1], dim=self.ch_axis, interpolation="lower")
+
+        # if any of the channels have 0s, we ignore that channel for this calculation
+        ch_no_zeros: torch.Tensor = (comp_quantile != torch.tensor([0])) | (cent_quartile != torch.tensor([0]))
```

Comment:
can we have a more descriptive name? like `any_zero_quantile_value`, also should `comp_quantile` and `cent_quantile` be vectors?